swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -519,3 +545,19 @@ extension WorkflowNode.SubtreeManager {
         }
     }
 }
+
+// MARK: - Side Effects
+
+extension WorkflowNode.SubtreeManager {
+    internal class SideEffectLifetime {
+        fileprivate let lifetime: Lifetime
+
+        fileprivate init() {
+            self.lifetime = Lifetime()
+        }
+
+        deinit {
+            lifetime.end()
```

Comment:
Might be good to leave a comment like `// Explicitly end the lifetime in case someone retained it from outside` since that question came up elsewhere and it’s a little subtle.